gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -747,6 +763,55 @@ private void submitJob(DagNode<JobExecutionPlan> dagNode) {
       }
     }
 
+    private void checkQuota(DagNode<JobExecutionPlan> dagNode) throws IOException {
```

Comment:
Can we distinguish between a normal job failure vs failure due to quota exceeding?

Reply:
How would we distinguish? Do you mean by throwing a different type of exception?